crates/matrix-sdk-base/src/read_receipts.rs

```diff
@@ -1547,4 +1560,56 @@ mod tests {
         // And the active receipt is the implicit one on my event.
         assert_eq!(read_receipts.latest_active.unwrap().event_id, event_id!("$6"));
     }
+
+    #[test]
+    fn test_compute_unread_counts_with_threading_enabled() {
+        fn make_event(user_id: &UserId, thread_root: &EventId) -> TimelineEvent {
+            EventFactory::new()
+                .text_msg("A")
+                .sender(user_id)
+                .event_id(event_id!("$ida"))
+                .in_thread(thread_root, event_id!("$latest_event"))
+                .into_event()
+        }
+
+        let mut receipts = RoomReadReceipts::default();
+
+        let own_alice = user_id!("@alice:example.org");
+        let bob = user_id!("@bob:example.org");
+
+        // Threaded messages from myself or other users shouldn't change the
+        // unread counts.
+        receipts.process_event(
+            &make_event(own_alice, event_id!("$some_thread_root")),
+            own_alice,
+            true,
+        );
```

Comment:
Note that in this commit (2nd), `process_event` doesn't take a boolean parameter yet, so the test wouldn't compile if checked out on this specific commit. Consider removing the bool param here, or squashing commits 2 and 3 :)